### PR TITLE
Update Alluxio SDK cache documentation

### PR DESCRIPTION
**Pull request by apc999**
Add documentation on how to use the Alluxio SDK cache as a built-in cache for Presto.

screenshot:

screencapture-file-Users-binfan-projects-presto-prestodb-presto-docs-target-html-cache-builtin-html-2020-07-06-15_33_36

```
== RELEASE NOTE ==
Add documentation on how to use the Alluxio SDK cache as a built-in cache for Presto.
```

### CODE REVIEW

**Comment by apc999** [comment]
@jainxrohit @highker  Please take a look at the doc

**Comment by jainxrohit** [comment]
I think we need to separate alluxio as a service usage vs the default data caching functionality available in the Presto.
Let's keep the default Alluxio data caching functionality separately and keep it simple. So that users quickly looking to enable the data caching can refer to it for reference.

**Comment by apc999** [comment]
@jainxrohit Please take a look

File: presto-docs/src/main/sphinx/cache.rst [hunk 1]
```diff
@@ -5,4 +5,5 @@ Cache
 .. toctree::
     :maxdepth: 1
 
+    cache/library
```
Comment (jainxrohit) [inline presto-docs/src/main/sphinx/cache.rst:8]: library and alluxio, seems confusing section to me. How about we create section for local and service(remote)?
WDYT @apc999 ?

File: presto-docs/src/main/sphinx/cache/library.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+To provide Presto an independent, distributed cache service for reads/writes and
+full-featured cache policies, please refer to :doc:`/cache/alluxio`.
+
+
+Setup
+-----
+
+To turn on this Alluxio cache library is very easy:
+simply include the following configuration in ``etc/catalog/hive.properties``
+and restart Presto workers.
```
Comment (jainxrohit) [inline presto-docs/src/main/sphinx/cache/library.rst:31]: Please note that node selection strategy is a coordinator level config and would require a restart of coordinator to be in effect. I think in the blog I had categorized it between coordinator and worker.

File: presto-docs/src/main/sphinx/cache/library.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    hive.node-selection-strategy=SOFT_AFFINITY
+    cache.enabled=true
+    cache.type=ALLUXIO
+    cache.alluxio.max-cache-size=500GB
+    cache.base-directory=/tmp/alluxio-cache
+
+In the above example,
+
+* ``hive.node-selection-strategy=SOFT_AFFINITY`` instructs Presto scheduler to take data affinity into consideration when secheduling tasks to workers that enables meaningful data caching effectiveness.
```
Comment (jainxrohit) [inline presto-docs/src/main/sphinx/cache/library.rst:43]: We should clarify that without soft affinity local data caching would be disabled. We should also mention the default NO_PREFERENCE here.

Comment (jainxrohit) [reply presto-docs/src/main/sphinx/cache/library.rst:43]: Let's also mention following parameter which has huge impact of the worker affinity and cache effectiveness
Here are some other configurations which can useful:
Coordinator configuration (useful to configure the definition of a busy worker):
    Set max pending splits per task: node-scheduler.max-pending-splits-per-task
    Set max splits per node: node-scheduler.max-splits-per-node

File: presto-docs/src/main/sphinx/cache/library.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
+=====================
```
Comment (jainxrohit) [inline presto-docs/src/main/sphinx/cache/library.rst:1]: I would have liked to start by saying Presto supports local data cache and then introduce Alluxio. Not a strong opinion.

File: presto-docs/src/main/sphinx/cache.rst (hunk 1 above)
Comment (apc999) [reply presto-docs/src/main/sphinx/cache.rst:8]: local and service can work

File: presto-docs/src/main/sphinx/cache/library.rst (hunk 2 above)
Comment (apc999) [reply presto-docs/src/main/sphinx/cache/library.rst:31]: updated

File: presto-docs/src/main/sphinx/cache/library.rst (hunk 3 above)
Comment (apc999) [reply presto-docs/src/main/sphinx/cache/library.rst:43]: done

File: presto-docs/src/main/sphinx/cache/library.rst (hunk 4 above)
Comment (apc999) [reply presto-docs/src/main/sphinx/cache/library.rst:1]: added

**Comment by apc999** [comment]
@jainxrohit comments addressed. Please take a look

File: presto-docs/src/main/sphinx/cache/local.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
+=====================
```
Comment (jainxrohit) [inline presto-docs/src/main/sphinx/cache/local.rst:1]: Can we have presto local cache as an introduction then talk about alluxio local cache?

File: presto-docs/src/main/sphinx/cache/local.rst [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+    hive.node-selection-strategy=SOFT_AFFINITY
+    cache.enabled=true
+    cache.type=ALLUXIO
+    cache.alluxio.max-cache-size=500GB
+    cache.base-directory=/tmp/alluxio-cache
+
+In the above example configuration,
+
+* ``hive.node-selection-strategy=SOFT_AFFINITY`` instructs Presto scheduler to take data affinity
```
Comment (jainxrohit) [inline presto-docs/src/main/sphinx/cache/local.rst:42]: It's not clear which config belongs to the coordinator, and which ones to the worker. We should create separate blocks for coordinator and workers.

Comment (jainxrohit) [reply presto-docs/src/main/sphinx/cache/local.rst:42]: You can refer to the blog if you need more info.

File: presto-docs/src/main/sphinx/cache/local.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  into consideration when secheduling tasks to workers that enables meaningful data caching effectiveness.
+  This configuration property is defaul to ``NO_PREFERENCE`` and local data caching is only enabled when set to ``SOFT_AFFINITY``.
```
Comment (jainxrohit) [inline presto-docs/src/main/sphinx/cache/local.rst:44]: default*

File: presto-docs/src/main/sphinx/cache/local.rst [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+Presto supports caching input data locally with a built-in cache
+to reduce query latency using Hive Connector.
+This cache, based on Alluixo client-side cache library,
+utilizes the local storage (such as SSD) with configurable capacity and locations.
+To understand its internals and the benchmark results of latency improvement,
+please read this `article <https://prestodb.io/blog/2020/06/16/alluxio-datacaching>`_.
+Note that this is a **read-cache**, **completely transparent** from users
+and **fully managed** by individual Presto workers.
+To provide Presto an independent, distributed cache service for read/write workloads with
```
Comment (jainxrohit) [inline presto-docs/src/main/sphinx/cache/local.rst:21]: I am not sure if it is the correct place to talk about the Alluxio cache as a service. WDYT @highker

File: presto-docs/src/main/sphinx/cache/local.rst [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+  This configuration property is defaul to ``NO_PREFERENCE`` and local data caching is only enabled when set to ``SOFT_AFFINITY``.
+  Other configuration on coordinator that can impact data affinity includes
+  ``node-scheduler.max-pending-splits-per-task`` (the max pending splits per task) and
+  ``node-scheduler.max-splits-per-node`` (the max splits per node).
+* ``cache.enabled=true`` turns on the local cache and ``cache.type=ALLUXIO`` sets it to Alluxio.
+* ``cache.alluxio.max-cache-size=500GB`` sets storage space to be 500GB.
+* ``cache.base-directory=/tmp/alluxio-cache`` specifies a local directory ``/tmp/alluxio-cache``. Note that this Presto server must have both read and write permission to access this local directory.
+
+
+Monitoring
```
Comment (jainxrohit) [inline presto-docs/src/main/sphinx/cache/local.rst:53]: May be, we should talk about Cache Recovery in this guide?

**Review by jainxrohit — DISMISSED** [review]

**Review by beinan — APPROVED** [review]
lgtm

File: presto-docs/src/main/sphinx/cache/local.rst (hunk 6 above)
Comment (apc999) [reply presto-docs/src/main/sphinx/cache/local.rst:41]: already included in the doc

File: presto-docs/src/main/sphinx/cache/local.rst (hunk 9 above)
Comment (apc999) [reply presto-docs/src/main/sphinx/cache/local.rst:52]: this cache is fully transparent to users with the latest version.
not sure what the users need to be aware of on cache recovery.

**Pull request merged**